aws_advanced_python_wrapper/read_write_splitting_plugin.py

```diff
@@ -263,6 +263,11 @@ def _switch_to_reader_connection(self, hosts: Tuple[HostInfo, ...]):
                 self._is_connection_usable(current_conn, driver_dialect)):
             return
 
+        hostnames = [host_info.host for host_info in hosts]
+        if self._reader_host_info is not None and self._reader_host_info.host not in hostnames:
```

Comment:
We will hit this block whenever read_only is set to True. Are there performance concerns for this block? It is necessary to protect against the case where we have previously cached a reader instance connection but that reader instance was later removed from the custom endpoint. Without this check the read/write plugin would continue to use the cached reader connection to the now-blocked instance.